File: Programming Basics with JS/T06 - While Loop/lecture/08. Graduation.js
function graduation(input) {
    let student = input[0];
    let averageGrade = 0.0;
    let total = 0.0;
    let i = 1;
    let classes = 12;

    while (i <= classes) {
        let currentGrade = Number(input[i]);

        if (currentGrade < 4.00) {
            console.log(`${student} has been excluded at ${i} grade`);
            return;
        }
        total += currentGrade;
        i++;
    }
    averageGrade = total / classes;
    console.log(`${student} graduated. Average grade: ${averageGrade.toFixed(2)}`);
}
graduation(["Gosho", "5", "5.5", "6", "5.43", "5.5", "6", "5.55", "5", "6", "6", "5.43", "5"]);

File: Programming Basics with Python/T06 - While Loop/lecture/08. Graduation.py
student = input()

total_notes = 0
grade = 0
bad_notes = 0

while True:
    notes = float(input())

    if notes >= 4:
        total_notes += notes
        grade += 1
        if grade == 12:
            average = total_notes / 12
            print(f'{student} graduated. Average grade: {average:.2f} ')
            break

    elif notes < 4:
        bad_notes += 1
        if bad_notes == 2:
            print(f'{student} has been excluded at {grade + 1} grade ')
            break